### ISSUE TITLE
Bug with renderer canvas

### ISSUE THREAD

**Issue by erkinsergey** [ppete2/Leaflet.PolylineMeasure#8]
Hi, updated measure dont work if leaflet map has canvas as a renderer:
`var map = new L.Map('map', {
                           renderer: L.canvas()
                        });`

**Comment by wernerschka** [comment]
Sorry I've no idea what causes the problem. Browser's Console tells the following error message:

_"TypeError: t._pxBounds is undefined"_ in leaflet.js

Maybe you want to post the issue on Leaflet's Github page too.

**Comment by ppete2** [comment]
Sorry I've no idea what causes the problem. Browser's Console tells the following error message:

_"TypeError: t.pxBounds is undefined" in leaflet.js

Maybe you want to post the issue on Leaflet's Github page too.

**Comment by GustavAT** [comment]
In order to draw an object on the canvas its pixel bounds must be known. The measurement-line (variables _templine_ and _path_) is initialized with an empy latlng-array. Thus the polyline's __project_-functions fails to calculate its correct pixel bounds.

A quick (and maybe dirty?) workaround would be to check for invalid bounds in the canvas-renderer's __extendRedrawBounds_ function. The following code worked for me:

```javascript
L.Canvas.include({
    _extendRedrawBounds: function (layer) {
        if (!layer._pxBounds) return;
        var padding = (layer.options.weight || 0) + 1;
        this._redrawBounds = this._redrawBounds || new L.Bounds();
        this._redrawBounds.extend(layer._pxBounds.min.subtract([padding, padding]));
        this._redrawBounds.extend(layer._pxBounds.max.add([padding, padding]));
    }
});
```

A better solution would be to add the line once two coordinates are known.

**Comment by GustavAT** [comment]
Yes, you are correct.
The canvas-renderer's bug seems to be fixed in Leaflet 1.2.0 (see https://github.com/Leaflet/Leaflet/blob/master/src/layer/vector/Canvas.js#L205) so only users with Leaflet version < 1.2.0 and canvas renderer should apply the fix.
